src/rascal/structure_managers/structure_manager.hh

```diff
@@ -320,72 +325,114 @@ namespace rascal {
     }
 
     /**
-     * Attach a property to a StructureManager. A given calculated property is
-     * only reasonable if connected with a structure. It is also connected with
-     * a sanity check so that the naming of attached properties is unique. If a
-     * property with the desired `name` already exists, a runtime error is
-     * thrown.
+     * Helper function to check if a property with the specifier `name` has
+     * already been attached in the manager layer which invoked this function.
      */
-    void attach_property(const std::string & name,
-                         std::shared_ptr<PropertyBase> property) {
-      if (this->has_property(name)) {
-        std::stringstream error{};
-        error << "A property of name '" << name
-              << "' has already been registered";
-        throw std::runtime_error(error.str());
-      }
-      this->properties[name] = property;
+    inline bool has_layer_property(const std::string & name) const {
```

Comment:
I would not make this function as it seems redundant and you could use `count` instead of find to search for the name as I feel it is more readable.

Reply:
To be clear
AdatporNL < StructureManagerCenter > is the whole stack
AdatporNL is a layer and StructureManagerCenter is a layer.

If you ask for a property, it has first to check if the property exists layer by layer, therefore I don't see any redundancy in having the two functions `has_layer_property` and `has_stack_property`.

Reply:
`is_property_in_current_level`